lib/auth/auth_with_roles.go

```diff
@@ -1364,30 +1366,21 @@ func (a *ServerWithRoles) ListUnifiedResources(ctx context.Context, req *proto.L
 		if err != nil {
 			return nil, trace.Wrap(err)
 		}
-		baseContext := a.context
+
+		requestableAccessChecker, err := newResourceAccessChecker(*extendedAuthCtx, types.KindUnifiedResource)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+
```

Comment:
should this only be created if we want to include requestable?

before when we used the base context, it was set outside inorder to defer setting it "back to normal", but since we are creating some new object here, we can just put it in the if, right?

we should jsut set `resourceAccess.accessChecker` always, and a requestable checker if `req.IncludeRequestable` is true.

Reply:
It is the case. See the if below. One of the reasons for refactor, this context swap and defer logic was confusing.

If `req.UseSearchAsRoles || req.UsePreviewAsRoles || req.IncludeRequestable` then we always use `requestableAccessChecker`, but it's either the primary checker or the secondary one depending if this is `req.IncludeRequestable`. That's how it was implemented before. Kind of confusing but that's how it was.

In my subsequent PR I'll be adding `req.UseSearchAsRoles` to the `req.UseSearchAsRoles` in the line 1377 below.